content/custom_resources/glossary.md

```diff
@@ -0,0 +1,578 @@
+
+Because both the custom properties and the **execute** properties are identical, this
+will result in an error message similar to:
+
+```ruby
+ArgumentError
+-------------
+wrong number of arguments (0 for 1)
+```
+
+To prevent this behavior, use `new_resource.` to tell Chef Infra Client
+to process the properties from the core resource instead of the
+properties in the custom resource. For example:
+
+```ruby
+property :command, String, name_property: true
+property :version, String
+
+# Useful properties from the `execute` resource
+property :cwd, String
+property :environment, Hash, default: {}
+property :user, [String, Integer]
+property :sensitive, [true, false], default: false
+
+prefix = '/opt/languages/node'
+
+load_current_value do
+  current_value_does_not_exist! if node.run_state['nodejs'].nil?
+  version node.run_state['nodejs'][:version]
+end
+
+action :run do
+  execute 'execute-node' do
+    cwd new_resource.cwd
+    environment new_resource.environment
+    user new_resource.user
+    sensitive new_resource.sensitive
+    # gsub replaces 10+ spaces at the beginning of the line with nothing
+    command <<-CODE.gsub(/^ {10}/, '')
+      #{prefix}/#{new_resource.version}/#{new_resource.command}
+    CODE
+  end
+end
+```
+
+where:
+
+- `cwd new_resource.cwd`
+- `environment new_resource.environment`
+- `user new_resource.user`
+- `sensitive new_resource.sensitive`
+
+correctly use the properties of the **execute** resource and not the identically-named override properties of the custom resource.
+
+## property
+
+Use the `property` method to define properties for the custom resource.
+The syntax is:
+
+```ruby
+property :property_name, ruby_type, default: 'value', parameter: 'value'
+```
+
+where
+
+- `:property_name` is the name of the property
+- `ruby_type` is the optional Ruby type or array of types, such as
+  `String`, `Integer`, `true`, or `false`
+- `default: 'value'` is the optional default value loaded into the
+  resource
+- `parameter: 'value'` optional parameters
+
+For example, the following properties define `username` and `password`
+properties with no default values specified:
+
+```ruby
+property :username, String
+property :password, String
+```
+
+## ruby_type
+
+The property ruby_type is a positional parameter.
+
+Use to ensure a property value is of a particular ruby class, such as:
+
+- `true`
+- `false`
+- `nil`
+- `String`
+- `Array`
+- `Hash`
+- `Integer`
+- `Symbol`
+
+Use an array of Ruby classes to allow a value to be of more than one type. For example:
+
+```ruby
+property :aaaa, String
+property :bbbb, Integer
+property :cccc, Hash
+property :dddd, [true, false]
+property :eeee, [String, nil]
+property :ffff, [Class, String, Symbol]
+property :gggg, [Array, Hash]
+```
+
+## sensitive
+
+
+A property can be marked sensitive by specifying `sensitive: true` on
+the property. This prevents the contents of the property from being
+exported to data collection and sent to an Automate server.
```

Comment:
Or shown in the logs if the Chef Infra Client run fails, which is usually what people care about